electronics_lib/PriceGetter.py

```diff
@@ -27,24 +27,28 @@ class GeneratePrice(BaseBackend):
     # part_number -> [ [lower-bound_1, price_1], [lower-bound_2 (and the previous upperbound), price_2], ... ]
     PRICE_TABLE = Dict[str, List[Tuple[int, float]]] = None
 
-    def generate_price_table(self):
-        if GeneratePrice.PRICE_TABLE is not None:
+    @classmethod
+    def get_price_table(cls) -> Dict[str, List[Tuple[int, float]]]:
+        if GeneratePrice.PRICE_TABLE is None:
             cur_path = os.path.dirname(__file__)
             parts_library = os.path.relpath('resources/Pruned_JLCPCB SMT Parts Library(20220419).csv', cur_path)
             with open(parts_library, 'r', newline='') as csv_file:
                 csv_reader = csv.DictReader(csv_file, 'r')
                 for row in csv_reader:
                     full_price_list = row['Price']
                     price_and_quantity_groups = full_price_list.split(',')
-                    value: List[Tuple[int, float]]
-                    for i in range(len(price_and_quantity_groups)):
-                        temp = price_and_quantity_groups[i].split(':')
-                        quantity_range = temp[0].split('-')
-                        value.append((int(quantity_range[0]), float(temp[1])))
+                    value: List[Tuple[int, float]] = []
+                    for price_and_quantity in price_and_quantity_groups:
+                        price_and_quantity_list = price_and_quantity.split(':')
+                        assert len(price_and_quantity_list) == 2
+                        quantity_range = price_and_quantity_list[0].split('-')
+                        assert len(quantity_range) == 1 or len(quantity_range) == 2
+                        value.append((int(quantity_range[0]), float(price_and_quantity_list[1])))
 
                     # sorts each value for PRICE_TABLE by lower-bounds, which is the first int in the Tuple:
                     sorted(value)
                     GeneratePrice.PRICE_TABLE[row['LCSC Part']] = value
```

Comment:
I'm getting an error here & idk what it is (i tried searching for it but couldn't really understand what's going on): "Cannot find reference '[' in 'None' "

Reply:
I think you need to initialize `GeneratePrice.PRICE_TABLE` to an empty dict, otherwise it's trying to index into the default None. Though that error message is really pretty unhelpful...

Also consider seeing what mypy is telling you, it should be particularly helpful for potential type-based errors like this.